Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMFactory.cpp

```diff
@@ -51,20 +51,28 @@ RemoteLegacyCDMFactory::RemoteLegacyCDMFactory(WebProcess& webProcess)
 
 RemoteLegacyCDMFactory::~RemoteLegacyCDMFactory() = default;
 
+void RemoteLegacyCDMFactory::ref() const
+{
+    m_webProcess->ref();
+}
+
+void RemoteLegacyCDMFactory::deref() const
+{
+    m_webProcess->deref();
+}
+
 void RemoteLegacyCDMFactory::registerFactory()
 {
     LegacyCDM::clearFactories();
     LegacyCDM::registerCDMFactory(
-        [weakThis = WeakPtr { *this }] (LegacyCDM* privateCDM) -> std::unique_ptr<WebCore::CDMPrivateInterface> {
-            if (weakThis)
-                return weakThis->createCDM(privateCDM);
-            return nullptr;
+        [protectedThis = Ref { *this }] (LegacyCDM& privateCDM) -> std::unique_ptr<WebCore::CDMPrivateInterface> {
```

Comment:
I think this would result in memory leak, no? `registerCDMFactory` will add it to `installedCDMFactories()`, which is a `static NeverDestroyed`.

Reply:
it's owned by the WebProcess. It's a singleton

Reply:
ah, ok.